pkgs/applications/networking/instant-messengers/teams/default.nix

```
@@ -0,0 +1,132 @@
+{ lib
+, stdenv
+, fetchurl
+, patchelf
+, makeWrapper
+, dpkg
+, libuuid
+, nspr
+, nss
+, alsaLib
+, cups
+, expat
+, dbus
+, glib
+, cairo
+, pango
+, atk
+, at-spi2-atk
+, gtk3
+, gdk-pixbuf
+, libxcb
+, libX11
+, libXext
+, libXfixes
+, libXi
+, libXcursor
+, libXcomposite
+, libXdamage
+, libXrender
+, libXrandr
+, libXtst
+, libXScrnSaver
+# Runtime dependencies
+, coreutils
+, gawk
+, xdg_utils
+, libsecret
+, libgnome-keyring
+, systemd }:
+
+assert stdenv.hostPlatform.system == "x86_64-linux";
+
+let
+
+  version = "1.2.00.32451";
+
+  rpath = stdenv.lib.makeLibraryPath [
+    libuuid
+    nspr
+    nss
+    alsaLib
+    cups
+    expat
+    dbus
+    glib
+    cairo
+    pango
+    atk
+    at-spi2-atk
+    gtk3
+    gdk-pixbuf
+    libxcb
+    libX11
+    libXext
+    libXfixes
+    libXi
+    libXcursor
+    libXcomposite
+    libXdamage
+    libXrender
+    libXrandr
+    libXtst
+    libXScrnSaver
+  ];
+
+  # Teams loads these dynamically.
+  modulePath = stdenv.lib.makeLibraryPath [
+    systemd # for libudev
+    libsecret
+    libgnome-keyring
+  ];
```

I have been able to start Teams by adding this to the `modulePath`. I do not know if it is a good practice (Slack derivation seems to use it).

```suggestion
  ] + ":${stdenv.cc.cc.lib}/lib64";
```

---

Thanks, I added it to `rpath` like in the Slack derivation rather than `LD_LIBRARY_PATH`. Does that work equally well for you? For some reason I cannot reproduce the error.

---

Strangely, it does not work with the addition to `rpath`, but it does work with it added to `modulePath`. I'm not really sure why, I'm curious to understand why it behaves this way. I'll do some `strace` digging.